Added editor state (#7)
inkdropapp/api-docs: src/app/states/editor/page.mdx

export const metadata = {
  title: 'editor',
  description: 'An app state of the editor. This section explains how to fetch and manipulate the `editor` state.',
}

# editor

An [app state](/guides/flux-architecture) of the editor.
You can obtain the `editor` state and integrate it with React components.{{ className: 'lead' }}

---

## Data Structure

<Row>
  <Col>
    <Properties>

>       <Property name="changed" type="boolean | undefined">
>         `true` if the note has been changed. 
>         The editor will save the note if `changed` is true when it is closed.
>         Otherwise it will be`undefined` if it has not been changed yet since the app launched.
>       </Property>

Reviewer: ```suggestion
      <Property name="changed" type="boolean">
        `true` if the note has been changed. 
        The editor will save the note if `changed` is true when it is closed.
      </Property>
```
As far as I can tell, this property cannot be undefined
https://docs.inkdrop.app/reference/state-editor#changed-boolean

>       <Property name="needsUpdatingDate" type="boolean | undefined">

Reviewer: ```suggestion
      <Property name="needsUpdatingDate" type="boolean">
```
As far as I can tell, this property cannot be undefined
https://docs.inkdrop.app/reference/state-editor#needsupdatingdate-boolean

        Whether the note's `updatedAt` field needs to be updated or not.
      </Property>

>       <Property name="viewMode" type="preview | sideBySide | edit">

Reviewer: ```suggestion
      <Property name="viewMode" type="'preview' | 'sideBySide' | 'edit'">
```
Strings must be enclosed in quotes, otherwise Typescript will resolve them as type.
As it is, it is easy to understand, but this change also allows users to simply copy the type declaration and use it directly.

        The editor view mode which represents either `'preview'`, `'sideBySide'` or `'edit'`.
      </Property>
      <Property name="openingNoteId" type="string | undefined">
        When the app is trying to open a note, it indicates its' note ID, otherwise `undefined`.
      </Property>
      <Property name="readOnly" type="boolean | undefined">
        true if the editor is in read only mode.
      </Property>
      <Property name="savingNoteId" type="string | undefined">

>         When the app is saving the note, it indicates its' node ID, otherwise will be `undifined`.

Reviewer: ```suggestion
        When the app is saving the note, it indicates its' note ID, otherwise will be `undefined`.
```
A typo in the old documentation

export const metadata = {
  title: 'editor',
  description: 'An app state of the editor. This section explains how to fetch and manipulate the `editor` state.',
}

> # editingNote

Reviewer 2: should be `editor`

Author: Fixed

An [app state](/guides/flux-architecture) of the editor.
You can obtain the `editor` state and integrate it with React components.{{ className: 'lead' }}

---

## Data Structure

<Row>
  <Col>
    <Properties>
      <Property name="changed" type="boolean | undefined">
        `true` if the note has been changed. 
        The editor will save the note if `changed` is true when it is closed.
        Otherwise it will be`undefined` if it has not been changed yet since the app launched.
      </Property>
      <Property name="needsUpdatingDate" type="boolean | undefined">
        Whether the note's `updatedAt` field needs to be updated or not.
      </Property>

>       <Property name="viewMode" type="EditorViewMode">

Author: @craftzdog This is the only one I am a little unsure of in the context of Typescript. Can you confirm if this is right or not?

Reviewer 2: It can be fine as is, but more precisely, it should be `'preview' | 'sideBySide' | 'edit'`. Can you change it?

        The editor view mode which represents either `'preview'`, `'sideBySide'` or `'edit'`.
      </Property>
      <Property name="openingNoteId" type="string | undefined">

>         When the app is trying to open a note, it indicates its' note ID, otherwise `undefined``.

Reviewer 2: duplicate backticks on the 'undefined'.

<img width="109" alt="Screenshot 2023-09-05 at 10 51 41" src="https://github.com/inkdropapp/api-docs/assets/1332805/fe7b8808-b20a-40a8-955f-f33521796016">

Author: Fixed

      </Property>
      <Property name="readOnly" type="boolean | undefined">
        true if the editor is in read only mode.
      </Property>
      <Property name="savingNoteId" type="string | undefined">

>         When the app is saving the note, it indicates its' node ID, otherwise will be `undifined``.

Reviewer 2: duplicate backticks

Author: Fixed

      </Property>

>       <Property name="lastError" type="Error | null">

Reviewer 2: `AssistiveError | null`

Author: Fixed

        The last error that occurred.
      </Property>   
    </Properties>
  </Col>
  <Col sticky>

>     Get the `editingNote` state manually:

Reviewer 2: `editor`

Author: Fixed

    ```js
    const { editor } = inkdrop.store.getState()
    ```

    Connect the state with your React component:

    ```js
    import { useSelector } from 'react-redux'
    const selector = ({ editor }) => editor
    const MyComponent = props => {
      const editor = useSelector(selector)
      // render
    }
    ```
  </Col>
</Row>

> [editor]: /data-access/editor

Reviewer 2: unused link. remove it

Author: Fixed
